tripy/tripy/backend/compiler_api.py

```diff
@@ -31,6 +32,136 @@
 from tripy.common.shape_bounds import ShapeBounds
 from tripy.frontend import Tensor, Trace
 from tripy.utils import json as json_utils
+from tripy.backend.mlir.utils import MLIRRuntimeClient
+
+
+@export.public_api(document_under="compiler")
+class Stream:
+    """
+    Represents a CUDA stream that can be used to manage concurrent operations.
+
+    This class is a wrapper around the underlying stream object, allowing management of CUDA streams.
+
+    Args:
+        default_stream (bool): If True, uses the default stream. If False, creates a new stream.
+
+    .. code-block:: python
+        :linenos:
+        :caption: Create new streams
+
+        stream_a = tp.Stream()
+        stream_b = tp.Stream()
+        default_stream = tp.Stream(default_stream=True)
+        assert stream_a != stream_b
+        assert default_stream != stream_b
+        stream_a.cleanup()
```

Comment:
Do we have to manually clean up streams? Can't the context manager handle that?

Reply:
This is only in the case where `Stream` is created without context manager. I can also just remove this case and we can rely only on context manager for Stream management.